### PR TITLE
print_exception can print to places that aren't stdout

### PR DESCRIPTION
**Pull request by FauxFaux**
This should allow me to .. show users errors.

I've left `print_exception` ignoring the error, as this is what `println!` does. I'd be happy for it to return the error.

### CODE REVIEW

File: vm/src/exceptions.rs [hunk 1]
```diff
@@ -25,32 +25,50 @@ fn exception_init(vm: &VirtualMachine, args: PyFuncArgs) -> PyResult {
 
 /// Print exception chain
 pub fn print_exception(vm: &VirtualMachine, exc: &PyObjectRef) {
+    let _ = write_exception(io::stdout(), vm, exc);
+}
+
+pub fn write_exception<W: Write>(
+    mut output: W,
+    vm: &VirtualMachine,
+    exc: &PyObjectRef,
```
Comment (coolreader18) [inline vm/src/exceptions.rs]: This could be `io::Result<()>`

File: vm/src/exceptions.rs [hunk 2]
```diff
@@ -25,32 +25,50 @@ fn exception_init(vm: &VirtualMachine, args: PyFuncArgs) -> PyResult {
...
     }
-    print_exception_inner(vm, exc)
+    print_exception_inner(output, vm, exc)
 }
 
-fn print_source_line(filename: &str, lineno: usize) {
+fn print_source_line<W: Write>(
+    mut output: W,
+    filename: &str,
+    lineno: usize,
```
Comment (coolreader18) [inline vm/src/exceptions.rs]: And here

File: vm/src/exceptions.rs [hunk 3]
```diff
@@ -59,15 +77,21 @@ fn print_source_line(filename: &str, lineno: usize) {
...
     }
+
+    Ok(())
 }
 
 /// Print exception occurrence location from traceback element
-fn print_traceback_entry(vm: &VirtualMachine, tb_entry: &PyObjectRef) {
+fn print_traceback_entry<W: Write>(
+    mut output: W,
+    vm: &VirtualMachine,
```
Comment (coolreader18) [inline vm/src/exceptions.rs]: And here, etc

**Comment by FauxFaux** [comment]
I switched to the `io::Result` alias. I left it out on purpose (I don't like those Result typedefs), but we may as well be consistent!

**Comment by coolreader18** [comment]
Thanks for contributing!

**Pull request merged**